jbmc/src/java_bytecode/generic_parameter_specialization_map_keys.cpp

```diff
@@ -80,40 +80,42 @@ void generic_parameter_specialization_map_keyst::insert_pairs_for_pointer(
   const pointer_typet &pointer_type,
   const typet &pointer_subtype_struct)
 {
-  if(is_java_generic_type(pointer_type))
+  if(!is_java_generic_type(pointer_type))
+    return;
+  // The supplied type must be the full type of the pointer's subtype
+  PRECONDITION(
+    pointer_type.subtype().get(ID_identifier) ==
+    pointer_subtype_struct.get(ID_name));
+
+  // If the pointer points to:
+  // - an incomplete class or
+  // - a class that is neither generic nor implicitly generic (this
+  //   may be due to unsupported class signature)
+  // then ignore the generic types in the pointer and do not add any pairs.
+  // TODO TG-1996 should decide how mocking and generics should work
+  //  together. Currently an incomplete class is never marked as generic. If
```

Comment:
:pick: An additional indent of 2 might be nicer than 1? (Not sure if we have a standard for this.) :house: